Formula/bitcoind.rb

```diff
@@ -0,0 +1,49 @@
+class Bitcoind < Formula
+  desc "Decentralized, peer to peer payment network"
+  homepage "https://bitcoin.org/"
+  url "https://github.com/bitcoin/bitcoin/archive/v0.14.1.tar.gz"
+  sha256 "4391dbf8fa9683f17c3b03feac429c1f3d71dcc6c0dab7d01733519880ea9834"
+
+  head do
+    url "https://github.com/bitcoin/bitcoin.git"
+  end
+
+  option "with-gui", "Build the GUI client (requires Qt5)"
+
+  depends_on :macos => :lion
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on "pkg-config" => :build
+  depends_on :xcode => :build
+  depends_on "berkeley-db@4"
+  depends_on "boost"
+  depends_on "openssl"
+  depends_on "libevent"
+  depends_on "miniupnpc" => :recommended
```

Comment:
In according to this [tutorial](https://github.com/bitcoin/bitcoin/blob/master/doc/build-unix.md#miniupnpc), `upnp` is optional feature, but it can useful for some cases.